services/sn_analysis_service/app.py

```diff
@@ -0,0 +1,138 @@
+
+    We expect the `inputs` dictionary to have the following keys:
+       - source_name: the name of the model to fit to the data
+       - fix_z: whether to fix the redshift
+       - photometry: the photometry to fit to the model (in csv format)
+       - redshift: the known redshift of the object
+
+    TODO in the next PR: send the results back to the webhook
+
+    """
+    data_dict = await dd
+
+    analysis_parameters = data_dict["inputs"].get("analysis_parameters", {})
+    analysis_parameters = {**default_analysis_parameters, **analysis_parameters}
+
+    source_name = analysis_parameters.get("source_name")
+    fix_z = analysis_parameters.get("fix_z") in [True, "True", "t", "true"]
+
+    data = Table.read(data_dict["inputs"]["photometry"], format='ascii.csv')
+    data.rename_column('mjd', 'time')
+    data.rename_column('filter', 'band')
+    data.rename_column('magsys', 'zpsys')
+
+    data['flux'].fill_value = 1e-6
+    data = data.filled()
+    data.sort("time")
+
+    redshift = Table.read(data_dict["inputs"]["redshift"], format='ascii.csv')
+    z = redshift['redshift'][0]
+
+    model = sncosmo.Model(source=source_name)
+
+    if fix_z and z is not None:
+        model.set(z=z)
+        bounds = {'z': (z, z)}
+    else:
+        bounds = {'z': (0.01, 1.0)}
+
+    # run the fit
+    result, fitted_model = sncosmo.fit_lc(
+        data,
+        model,
+        model.param_names,
+        bounds=bounds,
+    )
+    log(f"Number of χ² function calls: {result.ncall}")
+    log(f"Number of degrees of freedom in fit: {result.ndof}")
+    log(f"χ² value at minimum: {result.chisq}")
+    log(f"model parameters: {result.param_names}")
+    log(f"best-fit values: {result.parameters}")
+    log(f"The result contains the following attributes:\n {result.keys()}")
+    log(f'{result["success"]=}')
+
+    if result["success"]:
+
+        with tempfile.NamedTemporaryFile(
+            suffix=".png", prefix="snplot_", dir="/tmp/"
+        ) as f:
+            _ = sncosmo.plot_lc(
+                data,
+                model=fitted_model,
+                errors=result.errors,
+                model_label=source_name,
+                zp=23.9,
+                figtext=data_dict["resource_id"],
+                fname=f.name,
+            )
+            f.seek(0)
+            plot_data = io.BytesIO(f.read())
+            log('made lightcurve plot')
+
+        # make some draws from the posterior (simulating what we'd expect
+        # with an MCMC analysis)
+        post = rng.multivariate_normal(result.parameters, result.covariance, 10000)
+        post = post[np.newaxis, :]
+        # create an inference dataset
+        dataset = az.convert_to_inference_data(
+            {x: post[:, :, i] for i, x in enumerate(result.param_names)}
+        )
+
+        # TODO in the next PR #3:
+        # return the dataset and the plot data
+        # by calling the webhook
+        return (dataset, plot_data)
```

Comment:
N.B. We're just returning here (and no one is listening). In the next PR we'll bundle up the results and call the SP webhook to store the results.